### PR TITLE
DDS-1218-subscriber

### PR DESCRIPTION
**Pull request by dmlond**
a sneakers worker to process a queue of file_version_ids and use the DdsMd5Reporter to report new md5 hashsums for them

### CODE REVIEW

File: spec/dds_md5_subscriber_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      it {
+        Sneakers.configure(connection: BunnyMock.new)
+        expect(DdsMd5Reporter).to receive(:new)
+          .with(
+            file_version_id: file_version_id,
+            user_key: ENV['USER_KEY'],
+            agent_key: ENV['AGENT_KEY'],
+            dds_api_url: ENV['DDS_API_URL']
+          ).and_return(mock_reporter)
+        expect(mock_reporter).to receive(:report_md5)
```
Comment (dmann) [inline spec/dds_md5_subscriber_spec.rb]: To dry out the tests a bit, the 8 lines above could go in a `before(:example)` block and reused for both success and error contexts. To generalize the `:report_md5` mock, use a return block:
```
expect(mock_reporter).to receive(:report_md5) { report_md5_response }
```
And just set the `report_md5_response` to `nil` or `raise expected_error` in each context.

**Review by dmann — APPROVED** [review]
One suggestion to DRY out tests, but feel free to ignore it and merge.

File: spec/dds_md5_subscriber_spec.rb (hunk 1 above)
Comment (dmlond) [reply spec/dds_md5_subscriber_spec.rb]: I will try this again, but for some reason I tried running that in a before block at some point and the test failed. When I moved it into the it block, it passed. Not sure why.

Comment (dmlond) [reply spec/dds_md5_subscriber_spec.rb]: done

**Pull request merged**